w_common/lib/src/common/disposable.dart

```diff
@@ -552,12 +553,11 @@ class Disposable implements _Disposable, DisposableManagerV7, LeakFlagger {
     return completer;
   }
 
-  void manageDisposable(Disposable disposable) {
-    // ignore: unnecessary_null_comparison
+  void manageDisposable(Disposable? disposable) {
     if (disposable == null) {
       return;
     }
-    _throwOnInvalidCall('manageDisposable', 'disposable', disposable);
+    _throwOnInvalidDisposable('manageDisposable', 'disposable', disposable);
```

Comment:
**Question**: Is there a reason you need to add this new function? I took a look at the existing one and it seems like it would still work for us in this situation.

Reply:
The existing one does a null check on the incoming value and is used for non-Disposable values as well.

Reply:
We're doing a null check right above this line so we won't call that function if disposable is null. So it looks like we could still use the existing function. Were you seeing test errors or something using the existing one?

Reply:
Fair - I didn't have a 'good' reason for adding a new function and agree, adding a null check while preserving the original function usage is totally fine.